internal/controller/merge.go

```diff
@@ -160,10 +160,18 @@ func mergeIncomingFilters(curr, toMerge []frr.IncomingFilter) []frr.IncomingFilt
 	mergedIn := map[string]*frr.IncomingFilter{}
 	for _, a := range all {
 		f := a
-		_, found := mergedIn[f.Prefix]
-		if found {
+		curr, found := mergedIn[f.Prefix]
+		if !found {
+			mergedIn[f.Prefix] = &f
 			continue
 		}
+
+		if f.LEThan == 0 || curr.LEThan > f.LEThan {
```

Comment:
nit: `== 0` is unecessary

Reply:
yes and no. If I remove the == 0, I need to think.
If I leave it, I know that when unset, I set it.